### ISSUE TITLE
Automatic Type Checking: view but don't block

### ISSUE THREAD

**Issue by KevinMGranger** [konveyor/pelorus#708]
We've seen a few issues where type checking would have helped catch them before a runtime failure. I've been using type checking as it's built in to the vs code python extension.

We should add a type check as a pre-commit and CI step. While we're just starting to use it, it should just be a warning, not an error. We may decide to move towards errors some day. Even if it's just for certain files, for starters.

We do need to decide between MyPy and Pyright though.

|pyright | mypy |
|-------|-------|
| - more dependencies (I think it needs node.js as well as python) | + easier environment set up  |
 | + more featureful, on the forefront of some type checking evolution | - less featureful |
| + the default choice for the vs code python extension | - not the default |

I lean towards pyright personally, but we can experiment.

**Comment by mateusoliveira43** [comment]
Really like the idea of type checking! 
About the VS Code comparison, I believe it should not be part of discussion. But to use `mypy` with it, I just add
```json
    "python.linting.mypyEnabled": true,
    "python.linting.mypyArgs": [
        "--show-column-numbers",
        "--strict",
    ],
```
to `settings.json` (and `mypy` should be installed in the virtual environment to it to work).

**Comment by KevinMGranger** [comment]
I want to focus on a good development experience, so I think defaults matter. That's the only reason I brought up VS Code. But it is definitely easy to turn on mypy instead!

**Comment by KevinMGranger** [comment]
Closed with #710 and #713.
